### DISCUSSION TITLE
Ability to disable debug name

### DISCUSSION THREAD

**Discussion by ashley-taylor** [mobxjs/mobx#2728] [Ideas]
We have a rather large data structure.
Running the chrome heap dump we notice that the mobx debug names represent about 13% of our heap.
Would be nice if you could disable the names.

**Comment by danielkcz** [comment]
Um, why are you doing heap dump in development? Is it really useful for anything? I am sure many other libraries will pollute that output as well.

**Reply by ashley-taylor** [reply]
Its a production heap dump. every observable object / value looks to get a unique debug key.

**Reply by danielkcz** [reply]
Could you provide us with an example heap that has this along with the source code? It doesn't need to be from your app, just try some minimal approach that shows what's the issue.

**Comment by mweststrate** [comment]
Just checked, we do indeed completely unnecessarily generate unique debug
id at runtime in prod builds. That'd be a good thing to fix. PRs welcome!

**Comment by iChenLei** [comment]
@ashley-taylor mobx v6.1.6 will solve this problem #2780

**Reply by ashley-taylor** [reply]
Awesome. 
Have tested with the latest version and it has reduced our memory usage by around 7% Fantastic.
Spotted what looked up to another 5% but looks like the latest commit also address that.

**Reply by iChenLei** [reply]
Yes, v6.1.7 will be better. But we still need your feedback about this issue after the latest version release. cause we don't have real world test case for 100% coverage. Thanks, @ashley-taylor

**Reply by iChenLei** [reply]
Hi. mobx v6.1.7 is released, Can you do the same test as before? Thanks, your feedback is very useful for us to make `mobx` better.

**Reply by ashley-taylor** [reply]
Have just tested with v6.1.7 and can confirm it removes another 5% of our heap. 12% all up since v6.1.5 awesome.
Thanks a lot.
